### ISSUE TITLE
Addon's App Exports Getting Merged into Parent App's namespace (and not Engine's namespace)

### ISSUE THREAD

**Issue by ryanone** [ember-engines/ember-engines#489]
I have an app that defines an engine, and the engine depends on [ember-models-table 1.13.0](https://github.com/onechiporenko/ember-models-table/tree/a74a35b505180c3beb772c7b2ce9994eaa55a927).  The odd thing I'm noticing is that when I build, the [templates for the ember-models-table components](https://github.com/onechiporenko/ember-models-table/tree/a74a35b505180c3beb772c7b2ce9994eaa55a927/app/templates/components/models-table) (under the app directory)...when I look at their corresponding AMD modules, they're under the namespace of the parent app, as opposed to the engine which is actually directly depending on ember-models table.

I have a repo with a [pretty barebones Ember app](https://github.com/ryanone/ember-engines-issue) illustrating this.  The pertinent parts of the dependency graph are:

* parent app (ember-engines-issue)
** in-repo engine (lorem-ipsum-engine)
*** [ember-models-table 1.13.0](https://github.com/onechiporenko/ember-models-table/tree/a74a35b505180c3beb772c7b2ce9994eaa55a927)

When I run `ember build` and inspect the dist/assets/ember-engines-issue.js (JS file of the parent app), I notice that the [component templates](https://github.com/onechiporenko/ember-models-table/tree/a74a35b505180c3beb772c7b2ce9994eaa55a927/app/templates/components/models-table) from ember-models-table (all defined in the app/ directory) have AMD modules that are in the namespace of the parent app:

```
// Shouldn't this be in the namespace of the engine instead (ex. 'lorem-ipsum-engine/templates/components/models-table/table-footer')?
define("ember-engines-issue/templates/components/models-table/table-footer" ...
```

The result is that `{{partial}}` calls to these templates will fail (go to `/lorem-ipsum-engine/users` to see the browser error in the console), because they're in the namespace of the parent app.  One workaround I've had to implement is modifying the engine's resolver, by adding a module name lookup pattern function.

Can you confirm if this behavior is expected, or if this is actually an issue? Thanks!

**Comment by trentmwillis** [comment]
Sounds like an issue to me, though I'm unsure why it is happening. It looks like the link to the repro issue is not working, maybe the repo is private and needs to be made public?

**Comment by ryanone** [comment]
Thanks for responding @trentmwillis - sorry about that.  I made the repo public now.  Hopefully that unblocks you on your end.

**Comment by trentmwillis** [comment]
I checked out the repo. It looks some files from the addon are included in the Engine's namespace, just not the templates. I think this is because we never invoke the `addon-templates` tree from within `EngineAddon`. @rwjblue could verify that.

As for the addon code being in the app namespace, that's expected since the application has [`ember-models-table` listed as a dependency](https://github.com/ryanone/ember-engines-issue/blob/a282d2d0aeb27b8bda83d9bff91555d7e161533d/package.json#L36).

**Comment by ryanone** [comment]
Regarding the ember-models-table being merged into the app namespace, that makes sense given how in that app, the app directly depends on ember-models-table.  Should that be the case though, if ember-models-table is a *transitive* dependency?

FWIW, I created a second Ember app with a standalone/external engine, and was able to reproduce the issue.  In this case, `ember-models-table` is a transitive dependency, and so the dependency graph is as follows:

* [ryanone-app-with-engines](https://github.com/ryanone/ryanone-app-with-engines)
  * [ryanone-standalone-engine](https://github.com/ryanone/ryanone-standalone-engine)
    * ember-models-table

Run the app and go to `/standalone/users` and you'll see the error I initially mentioned, in the console.
